tests/Aspire.Hosting.PostgreSQL.Tests/PostgresFunctionalTests.cs

```diff
@@ -0,0 +1,366 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Npgsql;
+using Polly;
+using Xunit;
+
+namespace Aspire.Hosting.PostgreSQL.Tests;
+
+public class PostgresFunctionalTests
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyPostgresResource()
+    {
+        var builder = TestDistributedApplicationBuilder.Create();
```

Comment:
1. We should be using `TestDistributedApplicationBuilder.CreateWithTestContainerRegistry()` so we don't hit dockerhub during CI.
2. We should also add `builder.Services.AddLogging(b => b.AddXunit(testOutputHelper));` so the app logs get outputted to the test logs.